Xamarin.Essentials/SecureStorage/SecureStorage.shared.cs

```diff
@@ -8,29 +8,45 @@ public static partial class SecureStorage
         // Special Alias that is only used for Secure Storage. All others should use: Preferences.GetPrivatePreferencesSharedName
         internal static readonly string Alias = $"{AppInfo.PackageName}.xamarinessentials";
 
-        public static Task<string> GetAsync(string key)
+        // Special alias needed to used grouping
+        internal static string GetAlias(string accessGroup)
+           => string.IsNullOrWhiteSpace(accessGroup) ? Alias : $"{accessGroup}.{Alias}";
```

Comment:
Hmm, do we really want to use `$"{accessGroup}.{Alias}"` here? I'm just thinking if I was writing my app I might expect the string to be exactly what I pass in.  I'm torn of course because adding the Alias does keep it unique to Essentials' usage, but that might defeat the purpose of specifying an access group in the first place?

Reply:
It makes sense to me.  Although, perhaps there's a case that Alias should also be configurable. Perhaps if I have a suite of apps written in different tech stacks, yet they should have a common shared storage.

Reply:
Am I understanding correctly that in the case of app groups, this means that you would need to be provision a group in the format of:

     <group.groupName>.<packageName>.xamarinessentials

And wouldn't this also mean that sharing secure storage between two apps would be impossible because the package name (distinct between apps) that is added as part of Alias would always be different?

Reply:
@kiddailey is correct... Why do we have the package name in there? Aren't prefs already unique to the app?

Maybe the access group should be the override for that?

Also, since things are shared on ios between the app/extension, what is the equivalent for Android? Does it just use the fact that a watch app is the same as a mobile? Is there a communication mechanism?

Reply:
> @kiddailey is correct... Why do we have the package name in there? Aren't prefs already unique to the app?
> Maybe the access group should be the override for that?

Thanks for pointing this out @mattleibow.  You are correct that the access group should override, and it actually does that in the Android implementation.  iOS just needs to as well.  And the package name cannot be part of it.  See my post below detailing why.